src/mlpack/bindings/go/get_printable_type_impl.hpp

```diff
@@ -0,0 +1,151 @@
+    const typename boost::disable_if<util::IsStdVector<double>>::type*,
+    const typename boost::disable_if<data::HasSerialize<double>>::type*,
+    const typename boost::disable_if<arma::is_arma_type<double>>::type*,
+    const typename boost::disable_if<std::is_same<double,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  return "float64";
+}
+
+template<>
+inline std::string GetPrintableType<std::string>(
+    const util::ParamData& /* d */,
+    const typename boost::disable_if<util::IsStdVector<std::string>>::type*,
+    const typename boost::disable_if<data::HasSerialize<std::string>>::type*,
+    const typename boost::disable_if<arma::is_arma_type<std::string>>::type*,
+    const typename boost::disable_if<std::is_same<std::string,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  return "string";
+}
+
+template<>
+inline std::string GetPrintableType<size_t>(
+    const util::ParamData& /* d */,
+    const typename boost::disable_if<util::IsStdVector<size_t>>::type*,
+    const typename boost::disable_if<data::HasSerialize<size_t>>::type*,
+    const typename boost::disable_if<arma::is_arma_type<size_t>>::type*,
+    const typename boost::disable_if<std::is_same<size_t,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  return "float64";
+}
+
+template<>
+inline std::string GetPrintableType<bool>(
+    const util::ParamData& /* d */,
+    const typename boost::disable_if<util::IsStdVector<bool>>::type*,
+    const typename boost::disable_if<data::HasSerialize<bool>>::type*,
+    const typename boost::disable_if<arma::is_arma_type<bool>>::type*,
+    const typename boost::disable_if<std::is_same<bool,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  return "bool";
+}
+
+template<typename T>
+inline std::string GetPrintableType(
+    const util::ParamData& d,
+    const typename boost::enable_if<util::IsStdVector<T>>::type*,
+    const typename boost::disable_if<std::is_same<T,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  return "list of " + GetPrintableType<typename T::value_type>(d) + "s";
+}
+
+template<typename T>
+inline std::string GetPrintableType(
+    const util::ParamData& /* d */,
+    const typename boost::enable_if<arma::is_arma_type<T>>::type*,
+    const typename boost::disable_if<std::is_same<T,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  std::string type = "matrix";
+  if (std::is_same<typename T::elem_type, double>::value)
+  {
+    if (T::is_row || T::is_col)
+      type = "vector";
+  }
+  else if (std::is_same<typename T::elem_type, size_t>::value)
+  {
+    type = "int matrix";
+    if (T::is_row || T::is_col)
+      type = "int vector";
+  }
+
+  return type;
+}
+
+template<typename T>
+inline std::string GetPrintableType(
+    const util::ParamData& /* d */,
+    const typename boost::enable_if<std::is_same<T,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  return "categorical matrix";
```

Comment:
I will start the work on matrix with dataset info as soon as possible.Sorry for the delay.

Reply:
Sounds good, and no hurry---let me know what type you think is right in Go to do it with.  Probably that's the first thing to look at, but I don't think Go has a really robust data science toolkit, so maybe there is no "categorical matrix type" or anything like there is in Python via pandas.

Reply:
Here I think we can use a struct that contain both `[]bool` and `*mat.Dense` that works as a tuple just as Julia binding while checking the dimensions of Boolean array and gonum matrix because I didn't find any such categorical matrix type in golang without any additional support.

Reply:
Yeah, that seems reasonable to me. :+1:

Reply:
I will work on this part as soon as possible.

Reply:
This is also handled now, so I'll resolve this conversation. :+1: